ring/ring_poly.go

```diff
@@ -96,6 +96,27 @@ func (pol *Poly) Copy(p1 *Poly) {
 	}
 }
 
+// Equals returns true if the receiver Poly is equal to the provided other Poly.
+func (pol *Poly) Equals(other *Poly) bool {
```

Comment:
It should apply a modular reduction before checking equality

Reply:
Good point!